virttest/utils_disk.py

```diff
@@ -372,15 +373,22 @@ def create_partition_linux(session, did, size, start,
     :param start: partition beginning at start. e.g. 0M
     :param part_type: partition type, primary extended logical
     :param timeout: Timeout for cmd execution in seconds.
+    :return: the created partition.
     """
     size = utils_numeric.normalize_data_size(size, order_magnitude="M") + "M"
     start = utils_numeric.normalize_data_size(start, order_magnitude="M") + "M"
     end = str(float(start[:-1]) + float(size[:-1])) + size[-1]
     partprobe_cmd = "partprobe /dev/%s" % did
-    mkpart_cmd = 'parted -s "%s" mkpart %s %s %s'
-    mkpart_cmd %= ("/dev/%s" % did, part_type, start, end)
-    session.cmd(mkpart_cmd)
+    mkpart_cmd = 'parted -s "/dev/%s" mkpart %s %s %s print' % (did, part_type,
+                                                                start, end)
+    output = session.cmd(mkpart_cmd)
+    partition = did + re.findall(r'\s+(\d+)\s+\d+', output, re.M)[-1]
     session.cmd(partprobe_cmd, timeout=timeout)
+    if not wait.wait_for(
```

Comment:
partition might not be visible under `/dev/`, better to directly grep from `/proc/partitions` to see if partition exists, like:
```
# grep 'sda1' /proc/partitions
   8        1    1048576 sda1
```

Reply:
why might partition not be visible under /dev/?